src/components/node-list/node-list-items.js

```diff
@@ -331,3 +332,122 @@ export const getFilteredItems = createSelector(
     };
   }
 );
+
+// the following are selectors needed for the tree list data
+/**
+ * returns an array of the corresponding filtered nodes
+ * filtered nodes for each modular pipeline
+ */
+export const getFilteredModularPipelineNodes = createSelector(
+  [
+    getFilteredNodeItems,
+    getFilteredModularPipelineItems,
+    (state) => state.modularPipelineIds,
+    (state) => state.nodeTypeIDs,
+  ],
+  (
+    filteredNodeItems,
+    filteredModularPipelines,
+    modularPipelineIDs,
+    nodeTypeIDs
+  ) => {
+    const modularPipelineNodes = arrayToObject(modularPipelineIDs, () => []);
+    const { modularPipeline } = filteredModularPipelines;
+
+    // assumption: each node is unique and will only exist once on the flowchart, hence we are only taking
+    // the deepest nested modular pipeline as the node's modular pipeline
+    nodeTypeIDs.forEach((key) => {
+      filteredNodeItems[key]?.forEach((node, i) => {
+        if (node.modularPipelines.length > 1) {
+          node.modularPipelines = [
+            node.modularPipelines[node.modularPipelines.length - 1],
+          ];
+        }
+      });
+    });
+
+    // create a new field for the topmost / root pipeline
+    modularPipelineNodes['main'] = [];
```

Comment:
Maybe use `modularPipelineNodes.main` for consistency here?

Reply:
Thanks - I've incorporated it [here](https://github.com/quantumblacklabs/kedro-viz/blob/c8dbf269eb790a5ee2ddbb54398740bc296d31e9/src/components/node-list/node-list-items.js#L546)